docs/docs/adding-search-with-algolia.md

```diff
@@ -2,7 +2,604 @@
+  border-radius: ${props => props.theme.smallBorderRadius};
+  ${props => (props.collapse ? collapse : expand)};
+`
+
+export const Form = styled.form`
+  display: flex;
+  flex-direction: row-reverse;
+  align-items: center;
+`
+
+const list = css`
+  position: absolute;
+  right: 0;
+  top: calc(100% + 0.5em);
+  width: calc(4em + 40vw);
+  max-width: 30em;
+  box-shadow: 0 0 5px 0;
+  padding: 0.7em 1em 0.4em;
+  background: ${props => props.theme.white};
+  border-radius: ${props => props.theme.smallBorderRadius};
+  > * + * {
+    padding-top: 1em !important;
+    border-top: 2px solid ${props => props.theme.darkGray};
+  }
+  li + li {
+    margin-top: 0.7em;
+    padding-top: 0.7em;
+    border-top: 1px solid ${props => props.theme.lightGray};
+  }
+`
+
+const grid = css`
+  ul {
+    display: grid;
+    grid-template-columns: repeat(auto-fit, minmax(15em, 1fr));
+    grid-gap: 1em;
+    li {
+      padding: 0.3em 0.5em;
+      background: ${props => props.theme.veryLightGray};
+      border-radius: ${props => props.theme.smallBorderRadius};
+    }
+  }
+`
+
+export const HitsWrapper = styled.div`
+  display: ${props => (props.show ? `grid` : `none`)};
+  max-height: 80vh;
+  overflow: scroll;
+  ${props => (props.hitsAsGrid ? grid : list)};
+  * {
+    margin-top: 0;
+    padding: 0;
+  }
+  ul {
+    list-style: none;
+  }
+  mark {
+    color: ${props => props.theme.lightBlue};
+    background: ${props => props.theme.darkBlue};
+  }
+  header {
+    display: flex;
+    justify-content: space-between;
+    margin-bottom: 0.3em;
+    h3 {
+      color: ${props => props.theme.white};
+      background: ${props => props.theme.gray};
+      padding: 0.1em 0.4em;
+      border-radius: ${props => props.theme.smallBorderRadius};
+    }
+  }
+  h3 {
+    margin: 0 0 0.5em;
+  }
+  h4 {
+    margin-bottom: 0.3em;
+  }
+`
+
+export const By = styled.span`
+  font-size: 0.6em;
+  text-align: end;
+  padding: 0;
+`
+```
+
+Now we're almost done. 2 small steps remain. First, we need to put together a hit component for every type of result we want to display. In our example, these are blog posts and pages. And second, we need to call our `Search` component somewhere on our site. Here are the post and page hit components.
+
+```jsx
+// src/components/Search/PageHit.js
+import React from 'react'
+import { Highlight, Snippet } from 'react-instantsearch-dom'
+import { Link } from 'gatsby'
+
+const PageHit = clickHandler => ({ hit }) => (
+  <div>
+    <Link to={hit.slug} onClick={clickHandler}>
+      <h3>
+        <Highlight attribute="title" hit={hit} tagName="mark" />
+      </h3>
+    </Link>
+    <Snippet attribute="excerpt" hit={hit} tagName="mark" />
+  </div>
+)
+
+export default PageHit
+```
+
+```jsx
+// src/components/Search/PostHit.js
+import React from 'react'
+import { Highlight, Snippet } from 'react-instantsearch-dom'
+import { Link } from 'gatsby'
+
+const PageHit = clickHandler => ({ hit }) => (
+import React, { Fragment } from 'react'
```

Comment:
Did something get copy-pasted weird here? React is imported twice.

Reply:
Good catch! This is fixed now.